### DISCUSSION TITLE
Cursorless, voice in VSCode, and the future

### DISCUSSION THREAD

**Discussion by merriam** [cursorless-dev/cursorless#2259] [General]
What ate the thoughts of CursorLess functionality interacting with the [VS Code Speech Extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-speech)?   Currently, Microsoft
is providing a quick voice to speech, an entry box for its CoPilot, and a JSON file based snippet editor.  In time, one can 
expect Microsoft to make an inferior but widely free copy of anything in CursorLess.

It seems like CursorLess will need to have at least a quick way of switching to "Speech Extension gets next utterance" or other managment.   

This is an open question, meaning there are many different tracks to answers:

- Should CursorLess have interaction with VCSpeech?   For example, a "VCSpeech gets next utterance" or otherwise mitigating the 'fight for the microphone'?   Alternately, should Cursorless just subsume the VCSpeech functionality, which is "say", "say into a search box", and snippets?
- Should CursorLess think about a "don't Jeff me" license?
- Should CursorLess just get aquired by Microsoft?  Or Google?  How should this goal affect the direction of CursorLess?

The world did change.  How should Cursorless community and devlopment react?

**Comment by pokey** [comment]
I'm not sure I'm understanding the significance of the VSCode dictation feature. Macs already have dictation support built in; this feels fairly similar, no?

Re licensing / acquisition / whatever, I'm perfectly happy for Microsoft to take whatever they want; that's why it's open source 😊

Maybe let's discuss in one of our regular Cursorless Discord meet-ups?
